cwe_checker_rs/src/pcode/term.rs

```diff
@@ -410,10 +412,93 @@ pub struct Project {
 impl From<Project> for IrProject {
     /// Convert a project parsed from Ghidra to the internally used IR.
     fn from(project: Project) -> IrProject {
-        let program = Term {
+        let mut program: Term<IrProgram> = Term {
             tid: project.program.tid,
             term: project.program.term.into(),
         };
+        let register_map: HashMap<&String, &RegisterProperties> = project
+            .register_properties
+            .iter()
+            .map(|p| (&p.register, p))
+            .collect();
+        let mut zero_extend_tids: HashMap<String, Tid> = HashMap::new();
+        // iterates over definitions and checks whether sub registers are used
+        // if so, they are swapped with subpieces of base registers
+        for sub in program.term.subs.iter_mut() {
+            for blk in sub.term.blocks.iter_mut() {
+                let mut def_iter = blk.term.defs.iter_mut().peekable();
+                while let Some(def) = def_iter.next() {
+                    let peeked_def = def_iter.peek();
+                    match &mut def.term {
+                        IrDef::Assign { var, value } => {
+                            let zero_tid: Option<Tid> = value.process_sub_registers_if_necessary(
+                                Some(var),
+                                &register_map,
+                                peeked_def,
+                            );
+                            match zero_tid {
+                                Some(tid) => {
+                                    zero_extend_tids.insert(format!("{}", tid), tid);
+                                }
+                                _ => (),
+                            }
```

Comment:
You can combine the `let` with the `match` to an `if let Some(zero_tid) = ...` to make the code more concise.